stdlib/public/core/StringStorage.swift

```diff
@@ -711,6 +711,15 @@ final internal class __SharedStringStorage
     super.init()
     self._invariantCheck()
   }
+  
+  internal convenience init(
+    immortalCocoa ptr: UnsafePointer<UInt8>,
+    owner: AnyObject,
+    countAndFlags: _StringObject.CountAndFlags
+  ) {
+    self.init(immortal: ptr, countAndFlags: countAndFlags)
```

Comment:
Not clear to me that setting the immortal bit here is actually doing anything we need. It gives me a thing to check in `objCBridgeableObject`, but that's sort of abusing the flag, since you can imagine that we could have a shared string with an immortal owner that's not an NSString I guess?

Open to alternative approaches

Reply:
I think it's a fine use of the `immortal` flag. It is just used in this class's deinit:
```
  deinit {
    if (_owner == nil) && !immortal {
      start.deallocate()
    }
  }
```
It may be worth burning one of the bits in `countAndFlags` or `capacityAndFlags` instead of storing the `Bool` separately (see line 692.)